lib/sensors/MoistureSensor.cpp

```diff
@@ -0,0 +1,93 @@
+#include "mbed.h"
+#include "MoistureSensor.h"
+
+constexpr int Sensor_I2C_Address = 0x36 << 1;                               //MBED I2C uses 8 bit addressing, so addresses are left shifted by 1 (may need to be shifted by 2) 
+
+constexpr int Sensor_Status_Base = 0x00;                                    //Base address registers for different modules
+constexpr int Sensor_Moisture_Base = 0x0F;
+
+constexpr int Sensor_Status_HW_ID = 0x01;                                   //Function address register for the sensor's HW ID
+constexpr int Sensor_HW_ID_Code = 0x68;                                     //Expected value for sensor HW ID; NOTE: This is not the expected value as specified by the datasheet,
+                                                                            //it's just what I found through my own testing. It sometimes fluctuates on its own, so it's hard to trust :(
+
+constexpr int Sensor_Moisture_Function = 0x10;                              //Function address registers for moisture and temperature modules        
+constexpr int Sensor_Temp_Function = 0x04;  
+
+MoistureSensor::MoistureSensor(PinName sda, PinName scl) : i2c_(sda, scl){}
+
+bool MoistureSensor::Is_Initialized(){
+    if(this->Read_HW_ID() == Sensor_HW_ID_Code){                            //compare received HW ID Code to correct one
+        return true;
+    }
+
+    return false;
+}
+
+int MoistureSensor::Read_HW_ID(){
+    char cmd[2];
+    cmd[0] = Sensor_Status_Base;
+    cmd[1] = Sensor_Status_HW_ID;
+
+    char check[1];
+
+    i2c_.write(Sensor_I2C_Address, cmd, 2);                                 //initialize registers for checking device ID
+
+    i2c_.read(Sensor_I2C_Address, check, 1);                                //read device ID
+
+    return check[0];
+}
+
+uint16_t MoistureSensor::Read_Moisture(){
+    /*
+    if(!(this->Is_Initialized())){                                          //checks if device is initialized, returns 65534 if there is an issue
+        return 65534;                                                       //this is commented out because the HW_ID reading isn't consistent currently
```

Comment:
you mean that the HW ID is different between reads?

Reply:
Not exactly; it differed between different builds. For example, when I first flashed the code to test it, I was getting a value of 204 consistently, but after making some changes, it started reading out 188 instead. Now, it currently reads 104 (0x68, as noted in the comments) but I don't trust that it will stay consistent, given the... 'awkward' history.

Reply:
In those changes that you made, was it something to do with the addressing?

Reply:
Nope, but I figured out what the issue was: I didn't add a delay between writing and reading for checking HW_ID. I'm also going to double-check my delays for writing and reading in the moisture and temperature functions, to see if those values are affected.